pkg/ddl/ddl_worker.go

```diff
@@ -1181,9 +1174,29 @@ func (w *worker) processJobPausingRequest(d *ddlCtx, job *model.Job) (isRunnable
 	return true, nil
 }
 
-// runDDLJob runs a DDL job. It returns the current schema version in this transaction and the error.
-func (w *worker) runDDLJob(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, err error) {
-	defer tidbutil.Recover(metrics.LabelDDLWorker, fmt.Sprintf("%s runDDLJob", w),
+// runOneJobStep runs a DDL job *step*. It returns the current schema version in
+// this transaction, if the given job.Args has changed, and the error. The *step*
+// is defined as the following reasons:
+//
+// - TiDB uses "Asynchronous Schema Change in F1", one job may have multiple
+// *steps* each for a schema state change such as 'delete only' -> 'write only'.
+// Combined with caller transitOneJobStepAndWaitSync waiting for other nodes to
+// catch up with the returned schema version, we can make sure the cluster will
+// only have two adjacent schema state for a DDL object.
+//
+// - Some types of DDL jobs has defined its own *step*s other than F1 paper like
+// onRecoverTable, onLockTables, onCreateIndex, etc. Their purposes are various
+// but they make use of caller transitOneJobStep to persist job changes.
+//
+// - We may need to use caller transitOneJobStepAndWaitSync to make sure all
+// other node is synchronized to provide linearizability. So an extra job state
```

Comment:
please add an example, if i am a new comer, i will ask what is `extra job state`